GDJS/Runtime/AsyncTasksManager.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2013-2022 Florian Rival (Florian.Rival@gmail.com). All rights reserved.
+ * This project is released under the MIT License.
+ */
+namespace gdjs {
+  /**
+   * A manager for all asynchronous tasks.
+   * @see {@link RuntimeScene.getAsyncTasksManager}.
+   */
+  export class AsyncTasksManager {
+    /**
+     * Maps a task to the callback to be executed once it is finished.
+     */
+    private tasksWithCallback = new Array<{
+      asyncTask: AsyncTask;
+      callback: (runtimeScene: gdjs.RuntimeScene) => void;
+    }>();
+
+    /**
+     * Run all pending asynchronous tasks.
+     */
+    processTasks(runtimeScene: RuntimeScene): void {
+      for (let i = 0; i < this.tasksWithCallback.length; i++) {
+        const taskWithCallback = this.tasksWithCallback[i];
+        if (taskWithCallback.asyncTask.update(runtimeScene)) {
+          // The task has finished, run the callback and remove it.
+          taskWithCallback.callback(runtimeScene);
+          this.tasksWithCallback.splice(i--, 1);
```

Comment:
Thanks for the change! I had troubles reading that loop, I think  you can make it more readable with a while loop:

```js
      let taskIndex = 0;
      while (taskIndex < this.tasksWithCallback.length) {
        const taskWithCallback = this.tasksWithCallback[taskIndex];
        if (taskWithCallback.asyncTask.update(runtimeScene)) {
          // The task has finished, run the callback and remove it.
          taskWithCallback.callback(runtimeScene);
          this.tasksWithCallback.splice(taskIndex, 1);
        } else {
          taskIndex++
        }
      }
```

If it's ok with you, you can use it as well in `TaskGroup.update`

Reply:
Thaat's fine with me 👌